### DISCUSSION TITLE
ilogtail启动，有时会不收集数据。报错：logprocess queue is full, put modify event to event queue again。

### DISCUSSION THREAD

**Discussion by q121102302** [alibaba/ilogtail#834] [Help] [answered]
版本：1.3.1
报错日志信息：
![E6AF0112-8D12-4659-8558-08B20B76C066](https://user-images.githubusercontent.com/101638329/236606063-5a9151a8-7a00-44ca-8ec9-2abca0b6e6c0.png)
现象：
重启后有时正常。有时报错后 不收集数据。
配置：
ilogtail_config.json
`{
    "cpu_usage_limit": 3,
    "mem_usage_limit": 2048,
    "max_bytes_per_sec": 209715200,
    "process_thread_count": 4,
    "send_request_concurrency": 40,
    "max_read_buffer_size": 4194304,
    "buffer_file_num": 100,
    "buffer_file_size": 209715200,
    "buffer_file_path": ""
}`

**Comment by q121102302** [comment]
使用的场景：
k8s环境下直接部署，对logs下的目录做info日志的全监控，启动时的add a new watcher大约为40～60个的样子。但是实际有日志持续写入的大概也就10～20个文件。整体日志的写入速度并不算快  大约不超过500条/秒。但是每条日志相对会比较大一些。
采集配置：
[ilogtail_config.txt](https://github.com/alibaba/ilogtail/files/11405072/ilogtail_config.txt)
ilogtail日志:
<img width="1603" alt="image" src="https://user-images.githubusercontent.com/101638329/236721421-385183d4-23a2-4c61-920e-1d1412702c03.png">
<img width="1654" alt="image" src="https://user-images.githubusercontent.com/101638329/236721463-1db255c3-5b68-4ad0-a871-d162803b152c.png">

**Comment by linrunqi08** [comment]
采集配置和样例日志可以提供一下吗，方便我们复现一下，采集配置是指user_yaml_config.d下的配置文件

**Reply by q121102302** [reply]
采集配置：
[file_kafka.txt](https://github.com/alibaba/ilogtail/files/11405069/file_kafka.txt)
样例日志不怎么好提供。。我们内外分离 无法导出。。

**Reply by q121102302** [reply]
[micro-scm-mes-info-2023-05-04.0-20230505112056364414.log](https://github.com/alibaba/ilogtail/files/11405091/micro-scm-mes-info-2023-05-04.0-20230505112056364414.log)
这是之前遇到问题 导出的部分 这个情况较为特殊 error比较多   但是格式基本相同

**Comment by linrunqi08** [comment]
logtail_plugin的日志可以贴一下吗

**Comment by linrunqi08** [comment]
怀疑的原因：
1. processor_regex插件的正则比较复杂，解析性能可能会比较差
2. 发送的性能不够，反压到了log process

使用提供的样例数据和采集配置，处理插件没有复现出问题来，可以把kafka flusher换成flusher_stdout运行一段时间，用来确认是解析的问题还是发送的问题

**Reply by q121102302** [reply, marked as answer]
找到问题了  是原因2 我们的kafka 有一台机器假死了
![26932E99-8A40-41e6-AA2E-D30D1134E093](https://user-images.githubusercontent.com/101638329/237004181-f26f2b74-7e76-4a43-8738-0afff8f3dafe.png)
